include/envoy/network/io_handle.h

```diff
@@ -4,30 +4,108 @@
 
 #include "envoy/common/pure.h"
 
+#include "common/common/assert.h"
+
 namespace Envoy {
 namespace Network {
 
+// IoErrorCode::Again is used frequently. Define it to be a distinguishable address to avoid
+// frequent memory allocation of IoError instance.
+// If this is used, IoHandleCallResult has to be instantiated with a deleter that does not
+// deallocate memory for this error.
+#define ENVOY_ERROR_AGAIN reinterpret_cast<IoError*>(1)
```

Comment:
nit: Is it possible to use a constexpr here vs. a define?

Reply:
done

Reply:
not done. constexpr doesn't like reinterpret_cast.